### ISSUE TITLE
Frequent BSOD on Windows 10 Bash subssytem

### ISSUE THREAD

**Issue by AlbertoElias** [ipfs/go-ipfs#4606]
#### Version information:
go-ipfs version: 0.4.13-
Repo version: 6
System version: amd64/linux
Golang version: go1.9.2

#### Type:
Bug

#### Severity:
Critical

#### Description:
I keep on getting what seem like randoms BSOD.

There are 3 errors I see occurring:

- Running `ipfs daemon` in Windows 10, from the Bash subsystem. Sudden unexpected BSOD, sometimes when it seems to get stuck adding files, other times when a separate node seems to get stuck fetching files that I have pinned.
- Running `ipfs daemon` in Windows 10, from the Bash subsystem. BSOD when it seems to get stuck exiting after pressing `ctrl + c` twice.
- Running `ipfs daemon` in Windows 10, from the Bash subsystem. I have a pinned directory in that node. I then ssh to my Ubuntu VPS, and try to get that directory or even pin it. It doesn't seem to do anything, and suddenly my ssh connection closes. No BSOD here from Windows, just issues when fetching that directory. The directory can be visited just fine from a gateway.

(Edited to improve explanation)

**Comment by Stebalien** [comment]
What do you mean by "BSOD"? Linux doesn't have BSODs (i.e., it literally doesn't have a blue screen of death). Do you mean that linux is panicing (are the CapsLock and  ScrollLock keys flashing)? Is everything simply grinding to a halt (memory issues)?

**Comment by AlbertoElias** [comment]
Sorry, bad explanation, let me rephrase and also edit the above message. So there are 3 errors I see occurring:

- Running `ipfs daemon` in Windows 10, from the Bash subsystem. Sudden unexpected BSOD, sometimes when it seems to get stuck adding files, other times when a separate node seems to get stuck fetching files that I have pinned.
- Running `ipfs daemon` in Windows 10, from the Bash subsystem. BSOD when it seems to get stuck exiting after pressing `ctrl + c` twice.
- Running `ipfs daemon` in Windows 10, from the Bash subsystem. I have a pinned directory in that node. I then ssh to my Ubuntu VPS, and try to get that directory or even pin it. It doesn't seem to do anything, and suddenly my ssh connection closes. No BSOD here from Windows, just issues when fetching that directory. The directory can be visited just fine from a gateway.

**Comment by magik6k** [comment]
The fun thing is that I'm getting BSODs with windows version of gx with error codes saying that the kernel has ran out of heap space (AMD cpu).

@AlbertoElias Can you share more details? What error code(s) are you getting?

**Comment by AlbertoElias** [comment]
This is the BSOD I get. This happened again after closing the daemon. 

![img_20180201_222700](https://user-images.githubusercontent.com/913117/35708444-7f951f7c-07ae-11e8-95bc-ccd14ccac691.jpg)

**Comment by Stebalien** [comment]
So... I'm not sure if there's anything we can do here. This really looks like a Windows bug.

**Comment by alecbrick** [comment]
I'm experiencing the same issue here, under the same conditions. It's true that it could be a Windows bug, but I'm interested in figuring out exactly where it occurs. I'll update with more information if I can figure something out.

**Comment by djdv** [comment]
I'm surprised to hear this is still an issue, it was reported some time ago and was acknowledged by the WSL team.
https://github.com/Microsoft/WSL/issues/2669

Slightly off topic, can I ask why people are using the Linux binaries on Windows instead of the native Windows binaries?

**Comment by alecbrick** [comment]
Oh, hey, I literally just saw that issue.

Personally, I'm looking to get into contributing. Since I do most of my work on Windows, I'm using WSL for all my development.

**Comment by Stebalien** [comment]
So, this should now be fixed (according that issue). @alecbrick can you confirm with the latest windows version?

**Comment by djdv** [comment]
"Works on my machine"
![Untitled](https://user-images.githubusercontent.com/13862850/57713864-4db26700-7641-11e9-9734-f26fb38b961d.png)
